troposphere/static/js/actions/instance/stop.js

```diff
@@ -1,58 +1,88 @@
-            throw new Error("Missing instance");
-
-        var instance = params.instance,
-            instanceState = new InstanceState({
-                status_raw: "active - powering-off",
-                status: "active",
-                activity: "powering-off"
-            }),
-            originalState = instance.get("state"),
-            actionRequest = new InstanceActionRequest({
-                instance: instance
-            });
+/**
+ * Defines the `stop` steps for a single instance
+ */
+function handleOne(instance){
+    // if we've been passed a model via "selected resources" that
+    // is *not* an Instance, then just skip it - _move on_
+    if (!(instance instanceof Instance)) return;
 
+    let instanceState = new InstanceState({
+            status_raw: "active - powering-off",
+            status: "active",
+            activity: "powering-off"
+        }),
+        originalState = instance.get("state"),
+        actionRequest = new InstanceActionRequest({
+            instance: instance
+        });
+
+    instance.set({
+        state: instanceState
+    });
+
+    Utils.dispatch(InstanceConstants.UPDATE_INSTANCE, {
+        instance: instance
+    });
+
+    actionRequest.save(null, {
+        attrs: {
+            action: "stop"
+        }
+    }).done(function() {
+        // Consider "success" notification ...
         instance.set({
             state: instanceState
         });
+    }).fail(function(response) {
+        instance.set({
+            state: originalState
+        });
+        Utils.displayError({
+            title: "Your instance could not be stopped",
+            response: response
+        });
+    }).always(function() {
         Utils.dispatch(InstanceConstants.UPDATE_INSTANCE, {
             instance: instance
         });
-
-        actionRequest.save(null, {
-            attrs: {
-                action: "stop"
-            }
-        }).done(function() {
-            instance.set({
-                state: instanceState
-            });
-        }).fail(function(response) {
-            instance.set({
-                state: originalState
-            });
-            Utils.displayError({
-                title: "Your instance could not be stopped",
-                response: response
-            });
-        }).always(function() {
-            Utils.dispatch(InstanceConstants.UPDATE_INSTANCE, {
-                instance: instance
-            });
-            Utils.dispatch(InstanceConstants.POLL_INSTANCE_WITH_DELAY, {
-                instance: instance,
-                delay: 15*1000,
-            });
+        Utils.dispatch(InstanceConstants.POLL_INSTANCE_WITH_DELAY, {
+            instance: instance,
+            delay: 15*1000,
         });
+    });
+}
+
+
+export default {
+
+    stop: function(params) {
+        if (!params.resources)
+            throw new Error("Missing resources");
+        if (params.instance)
+            throw new Error("Deprecated parameter passed");
```

Comment:
I think this should be removed, it makes sense while developing the feature, but doesn't belong in the release.